internal/test/compilecheck/compile_check_test.go

```diff
@@ -12,21 +12,74 @@ import (
 	"io"
 	"os"
 	"path/filepath"
+	"strings"
 	"testing"
 
-	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 	"github.com/testcontainers/testcontainers-go"
 )
 
-var versions = []string{
-	"1.19",
-	"1.20",
-	"1.21",
-	"1.22",
-	"1.23",
-	"1.24",
-	"1.25",
+const mainGo = `package main
+
+import (
+	"fmt"
+
+	"go.mongodb.org/mongo-driver/v2/bson"
+	"go.mongodb.org/mongo-driver/v2/mongo"
+	"go.mongodb.org/mongo-driver/v2/mongo/options"
+)
+
+func main() {
+	_, _ = mongo.Connect(options.Client())
+	fmt.Println(bson.D{{Key: "key", Value: "value"}})
+}
+`
+
+const goMod = `module compilecheck
+
+go 1.19
+
+require go.mongodb.org/mongo-driver/v2 v2.1.0
+
+replace go.mongodb.org/mongo-driver/v2 => /mongo-go-driver
+`
+
+// goVersions is the list of Go versions to test compilation against.
+// To run tests for specific version(s), use the -run flag:
+//
+//	go test -v -run '^TestCompileCheck/go:1.19$'
+//	go test -v -run '^TestCompileCheck/go:1\.(19|20)$'
+var goVersions = []string{"1.19", "1.20", "1.21", "1.22", "1.23", "1.24", "1.25"}
```

Comment:
[optional] Suggest doing the versions 1 per line so we can add a comment for 1.19 and 1.25:

```go
var versions = []string{
	"1.19", // Minimum supported Go version for mongo-driver v2
	"1.20",
	"1.21",
	"1.22",
	"1.23",
	"1.24",
	"1.25", // Test suite Go Version
}
```